src/foam/core/reflow/DAOPrompt.js

```diff
@@ -165,17 +165,58 @@ foam.CLASS({
       adapt: function(o, n, p) {
         let oldAdapt = foam.dao.DAOProperty.ADAPT;
         if ( foam.String.isInstance(n) ) {
-          if ( this.scope[n] ) {
-            this.daoKey = n;
-            n = this.scope[n];
-          } else if ( this.scope[n + 'DAO'] ) {
-            this.daoKey = n + 'DAO';
-            n = this.scope[n + 'DAO'];
-          } else if ( this.__context__[n + 'DAO'] ) {
-            n =  n + 'DAO';
-          } else if ( n.endsWith('s') ) {
-            this.daoKey = n;
-            n = n.substring(0, n.length-1) + 'DAO';
+          // Handle dotted keys by traversing down the scope hierarchy
```

Comment:
We should make a FLOWDAOProperty that extends DAOProperty and standardize this behaviour so we don't need to repeat it all over the place.